Source/Applications/MiMD/Model/PRC002/ComplianceField.cs

```diff
@@ -105,49 +124,60 @@ public bool Evaluate(string value)
         /// </summary>
         /// <param name="value"> the value that is checked </param>
         /// <returns> whether <see cref="value"/> sattisfies the condition </returns>
-        public bool Evaluate(double value)
+        public bool Evaluate(double value, string PreVal)
         {
-            if (FieldType != "number")
+            double dynamicEvaluatedValue;
+            ExpressionContext context = new ExpressionContext();
+            context.Variables.Clear();
+
+            if (PreVal != null)
+                context.Variables["PreVal"] = double.Parse(PreVal);
+
+            try
+            {
+                IDynamicExpression e = context.CompileDynamic(Value.ToString());
+                object dynamicEvaluatedObject = e.Evaluate();
+                dynamicEvaluatedValue = Convert.ToDouble(dynamicEvaluatedObject as IConvertible);
+            }
+            catch
             {
                 return false;
             }
 
-            // Using Flee to evaluate the dynamic value
-            ExpressionContext context = new ExpressionContext();
-            IDynamicExpression e = context.CompileDynamic(value.ToString());
-            double dynamicEvaluatedValue = (double)e.Evaluate();
+
+            if (FieldType != "number")        
+                return false;
 
             if (Comparison == "IN")
             {
                 List<double> checks = Value.Split(';').Select(item => double.Parse(item)).ToList();
-                if (checks.Contains(dynamicEvaluatedValue))
+                if (checks.Contains(value))
                     return true;
                 return false;
             }
 
-            double fixedValue = double.Parse(Value);
 
-            if (Comparison == "=" && fixedValue == dynamicEvaluatedValue)
+            if (Comparison == "=" && value == dynamicEvaluatedValue)
                 return true;
-            if (Comparison == ">" && fixedValue < dynamicEvaluatedValue)
+            if (Comparison == ">" && value > dynamicEvaluatedValue)
                 return true;
-            if (Comparison == "<" && fixedValue > dynamicEvaluatedValue)
+            if (Comparison == "<" && value < dynamicEvaluatedValue)
                 return true;
-            if (Comparison == "<>" && fixedValue != dynamicEvaluatedValue)
+            if (Comparison == "<>" && value != dynamicEvaluatedValue)
                 return true;
 
             return false;
-        }
 
+        }
 
         /// <summary>
         /// Checks if a value satisfies the Condition on this Field
         /// </summary>
         /// <param name="value"> the value that is checked </param>
         /// <returns> whether <see cref="value"/> sattisfies the condition </returns>
-        public bool Evaluate(int value)
+        public bool Evaluate(int value, string PreVal)
```

Comment:
```suggestion
        public bool Evaluate(int value, intPreVal)
```

see above. for clarity I would make types match on these